Sources/DotLottie/Public/Player.swift

```diff
@@ -74,23 +74,143 @@ class Player {
+    }
+    
+    public func setConfig(config: Config) {
+        dotLottiePlayer.setConfig(config: config)
+    }
+    
+    public func config() -> Config {
+        return dotLottiePlayer.config()
+    }
+    
+    public func totalFrames() -> Float {
+        return dotLottiePlayer.totalFrames()
+    }
+    
+    public func setFrame(no: Float32) -> Bool {
+        return dotLottiePlayer.setFrame(no: no)
+    }
+    
+    public func currentFrame() -> Float {
+        return dotLottiePlayer.currentFrame()
+    }
+    
+    public func loopCount() -> Int {
+        return Int(dotLottiePlayer.loopCount())
+    }
+    
+    public func isLoaded() -> Bool {
+        return dotLottiePlayer.isLoaded()
+    }
+    
+    public func isPlaying() -> Bool {
+        return dotLottiePlayer.isPlaying()
     }
+    
+    public func isPaused() -> Bool {
+        return dotLottiePlayer.isPaused()
+    }
+    
+    public func isStopped() -> Bool {
+        return dotLottiePlayer.isStopped()
+    }
+    
+    public func isComplete() -> Bool {
+        return self.dotLottiePlayer.isComplete()
+    }
+    
+    public func play() {
+        let _ = dotLottiePlayer.play()
+        
+        self.setPlayerState(state: .playing)
+    }
+    
+    public func pause() {
+        let _ = dotLottiePlayer.pause()
 
-    func frame(no: Float32) {
-        dotLottiePlayer.frame(no: no)
+        self.setPlayerState(state: .paused)
     }
     
-    func currentFrame() -> Float32 {
-        return dotLottiePlayer.getCurrentFrame()
+    public func stop() {
+        let _ = dotLottiePlayer.stop()
+        
+        self.setPlayerState(state: .stopped)
     }
     
-    func duration() -> Float32 {
-        return dotLottiePlayer.getDuration()
+    public func resize(width: Int, height: Int) throws {
+        self.WIDTH = UInt32(width)
+        self.HEIGHT = UInt32(height)
+        
+        if (!dotLottiePlayer.resize(width: self.WIDTH, height: self.HEIGHT)) {
+            throw PlayerErrors.resizeError
+        }
+    }
+    
+    public func requestFrame() -> Bool {
+        let frame = dotLottiePlayer.requestFrame()
+
+        return self.setFrame(no: frame)
+    }
+       
+    public func duration() -> Float32 {
+        return dotLottiePlayer.duration()
     }
     
-    func clear() {
+    public func clear() {
         dotLottiePlayer.clear()
     }
+    
+    private func setPlayerState(state: PlayerState) {
+        DispatchQueue.main.async {
+            self.playerState = state
+        }
+    }
+    
+    func onLoad() {
+        self.setPlayerState(state: .loaded)
+    }
+    
+    func onLoop(loopCount: UInt32) {
+    }
+
+    func onComplete() {
+        self.setPlayerState(state: .paused)
+    }
+    
+    func onFrame(frameNo: Float) {
+    }
+
+    func onPause() {
+        self.setPlayerState(state: .paused)
+    }
+    
+    func onPlay() {
+        self.setPlayerState(state: .playing)
+    }
+    
+    func onRender(frameNo: Float) {
```

Comment:
@eharrison This method is so that Player respects the Observer protocol but for the moment no action needs to be taken inside onRender, same for the other methods